src/expandableCalendar/Context/Provider.tsx

```diff
@@ -89,24 +86,27 @@ const CalendarProvider = (props: CalendarContextProviderProps) => {
     animateTodayButton(currentDate);
   }, [todayBottomMargin, currentDate]);
 
+  /** Context */
+
   const _setDate = (date: string, updateSource: UpdateSources) => {
-    const updateState = (buttonIcon: number) => {
-      setPrevDate(currentDate);
-      setCurrentDate(date);
-      setUpdateSource(updateSource);
-      setButtonIcon(buttonIcon);
-    };
+    setPrevDate(currentDate);
```

Comment:
I wonder of these calls trigger multiple renders

Reply:
When I log the provider, I see one call after `setDate`.